### PR TITLE
PKG connector documentation

### LINKED ISSUE #108: Add documentation on PKG connector

### PR DESCRIPTION
**Pull request by NoB0**
Fixes #108

### CODE REVIEW

File: docs/source/pkg_connector.rst [hunk 1]
```diff
@@ -0,0 +1,28 @@
+PKG Connector
+=============
+
+This component is responsible for establishing a connection to the PKG and executing SPARQL queries against it.
+The SPARQL queries are generated based on the user's annotated statements using utility functions.
```
Comment (WerLaj) [inline docs/source/pkg_connector.rst:5]: Can we link the file with utility function here?

**Review by WerLaj — APPROVED** [review]
LGTM

File: docs/source/pkg_connector.rst (hunk 1 above)
Comment (NoB0) [reply docs/source/pkg_connector.rst:5]: Done.

File: docs/source/pkg_connector.rst [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+This component is responsible for establishing a connection to the PKG and executing SPARQL queries against it.
+The SPARQL queries are generated based on the user's annotated statements using utility functions (see :py:mod:`pkg_api.utils`).
+
+Features
+--------
+
+The PKG Connector has the following features:
+
+- Loading the PKG
```
Comment (kbalog) [inline docs/source/pkg_connector.rst:12]: Nit: perhaps combine "Loading and saving the PKG" as a single item

File: docs/source/pkg_connector.rst [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+- Binding the namespaces related to the PKG vocabulary (see complete list `here <https://iai-group.github.io/pkg-vocabulary/>`)
+- Executing SPARQL queries against the PKG
+- Saving the PKG
+
+The PKG Connector uses the `RDFLib <https://github.com/RDFLib/rdflib>` library to handle the PKG and execute the SPARQL queries. For this reason, we differentiate between two types of SPARQL queries: (1) queries to update the PKG, such as adding and removing statements, and (2) queries to retrieve information from the PKG.
```
Comment (kbalog) [inline docs/source/pkg_connector.rst:17]: Nit: "execute the SPARQL queries" => drop "the"

File: docs/source/pkg_connector.rst [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+The SPARQL queries are automatically generated based on the user's annotated statements and templates. 
+
+- **Insertion queries** for adding statements or preferences to the PKG, the templates are defined in :py:func:`pkg_api.utils.get_query_for_add_statement` and :py:func:`pkg_api.utils.get_query_for_add_preference` respectively.
+- **Deletion queries** for removing statements or preferences from the PKG, the templates are defined in :py:func:`pkg_api.utils.get_query_for_remove_statement` and :py:func:`pkg_api.utils.get_query_for_remove_preference` respectively. Additionally, a query to remove dangling objects is provided.
+- **Retrieval queries** for retrieving statements or preferences from the PKG, the templates are defined in :py:func:`pkg_api.utils.get_query_for_get_statements` and :py:func:`pkg_api.utils.get_query_for_get_preferences` respectively. For more flexibility, templates for retrieving statements and preferences based on specific criteria are also provided, see :py:func:`pkg_api.utils.get_query_for_conditional_get_statements` and :py:func:`pkg_api.utils.get_query_for_conditioned_get_preference`.
+
+All templates are defined in the `pkg_api.utils` module.
```
Comment (kbalog) [inline docs/source/pkg_connector.rst:28]: Is it possible to add a link to this module?

**Review by kbalog — APPROVED** [review]
LGTM

File: docs/source/pkg_connector.rst (hunk 3 above)
Comment (NoB0) [reply docs/source/pkg_connector.rst:17]: Done.

File: docs/source/pkg_connector.rst (hunk 4 above)
Comment (NoB0) [reply docs/source/pkg_connector.rst:28]: Done.

**Pull request merged**